src/main/java/views/add_edit_epense_views/EditExpenseV.java

```diff
@@ -102,27 +104,36 @@ public void actionPerformed(ActionEvent evt) {
                 if (recurExpense.getName().equals(selectedExpense)) {
                     // Set isRecurringCheckBox checked if the expense being edited was a recurring expense
                     isRecurringCheckBox.setSelected(true);}}
-        }
+        } else if (selectedExpense == null) {
+            JOptionPane.showMessageDialog(this, "Please select an expense you wish to edit.");}
 
-        if(selectedExpense == null){
-            JOptionPane.showMessageDialog( this, "Please select an expense you wish to edit.");
-        }else if(evt.getSource() == submit){
-            String selectedCategory = (String) categoryCombo.getSelectedItem();
-            boolean isRecurring = isRecurringCheckBox.isSelected();
-            ExpenseOD message = null;
-            try {
-                message = controller.expenseInMonth(nameInput.getText(), valueInput.getText(), selectedCategory,
-                        isRecurring, monthID, currSession, selectedExpense);
-                // Update Month Menu
+        if (evt.getSource() == categoryCombo) {
```

Comment:
I know we talked about minimizing the use of actionListeners, but I don't know how to make it work without one in this case. Please provide suggestions if any!

Reply:
my new commit in EditExpenseV is to fix another issue I found which is: When a user tries to edit an expense without selecting a category, the expense would automatically be assigned to the Other category since it is automatically selected. 

So to fix that problem, I decided to make sure the user inputs all required data before entering the use case. And what I did is to add the actionListener to categoryCombo and basically, it makes sure an action is performed. The conditions are added so that it is coordinated with EdotCategoryV's actionPerformed(). 

The issue in EditCategoryV that you mentioned(which is fixed!) is just a problem with the if-else statements such that certain ifs are not correctly accessed. (EditExpenseV had a similar problem in the same area but I also made changes to address the problem I mentioned above so it might be hard to tell)

Reply:
Though I don't think that would be a big problem. The point of having users to select a expense to edit is because the controller would take the input and add a new Expense if they don't. (similarly when in the EditCategoryV case, if a category is not selected, the controller will make a new Category). 

However, in this case, automatically selecting Other will not cause errors in the program: if they want to select Other, they do not have to touch the select Category, and the result will be what they want; if they want to select another Category, they will naturally click on the ComboBox. Moreover, the user can just edit the expense again if they make a mistake. 

The current implementation doesn't work properly, while the previous implementation (one before latest commit) works fine, so that's why I said to use the previous version.

Reply:
I fixed that problem just now since it's just a problem with the if-else statements(again!! :O) in EditExpenseV. You will see it in my next commit!

Reply:
Just so you know this part that your suggesting is not the reason why its not working lol